src/components/SubRequirement.vue

```diff
@@ -101,13 +141,26 @@ export default Vue.extend({
       required: false,
     },
     color: String,
-    isCompleted: Boolean
+    isCompleted: Boolean,
+    rostersFromLastTwoYears: Array as PropType<readonly String[]>
+  },
+  watch: {
+    toggleableRequirementChoice: {
+      immediate: true,
+      deep: true,
+      handler(updatedToggleableRequirementChoice) {
+        this.generateSubReqCoursesNotTakenArray();
+      }
+    }
   },
```

Comment:
I think this should not using `watch`, but instead let `subReqCoursesNotTakenArray` a computed property. You can see that `generateSubReqCoursesNotTakenArray` is almost a pure function, except that it imperatively change the `subReqCoursesNotTakenArray` data.

Reply:
@SamChou19815 I will make `subReqCoursesNotTakenArray` a computed property. I tried to also make `subReqCourseObjectsNotTakenArray` a computed property but I was entering into an infinite loop. So, I replaced the `watch` for `toggleableRequirementChoice` with a `watch` for `subReqCoursesNotTakenArray` that would call `getSubReqCourseObjects()`.